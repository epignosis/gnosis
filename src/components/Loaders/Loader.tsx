import { SerializedStyles } from "@emotion/react";
import React, { FC } from "react";
import { PulseLoader } from "react-spinners";
import { container } from "./styles";

export type LoaderSize = "md" | "lg";

export type LoaderProps = React.HTMLAttributes<HTMLDivElement> & {
  fullScreen?: boolean;
  size?: "md" | "lg";
};

const Loader: FC<LoaderProps> = ({ fullScreen = false, size = "lg", ...rest }) => (
  <div
    css={(theme): SerializedStyles => container(theme, { fullScreen, size })}
    data-testid="loader"
    {...rest}
    className={`loading-container${rest?.className ? ` ${rest?.className}` : ""}`}
  >
    <PulseLoader />
  </div>
);

export default Loader;
