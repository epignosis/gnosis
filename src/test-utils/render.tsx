import React, { FC, ReactElement, ComponentType } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options });

export * from "@testing-library/react";

export { customRender as render };
