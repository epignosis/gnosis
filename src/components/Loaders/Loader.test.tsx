import React from "react";
import Loader from "./Loader";
import { render } from "@test-utils/render";

describe("<Loader>", () => {
  it("renders correctly", () => {
    const { getByTestId } = render(<Loader />);
    const loader = getByTestId("loader");

    expect(loader).toBeInTheDocument();
    expect(loader).toBeVisible();
  });

  it("matches snapshot", () => {
    const { container } = render(<Loader id="main-loader" className="loader" size="lg" />);

    expect(container).toMatchSnapshot();
  });

  it("matches snapshot with fullscreen", () => {
    const { container } = render(<Loader fullScreen />);

    expect(container).toMatchSnapshot();
  });
});
