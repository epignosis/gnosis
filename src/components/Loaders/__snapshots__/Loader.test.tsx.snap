// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader> matches snapshot 1`] = `
<div>
  <div
    class="loading-container loader css-1hn9l2b-container-container"
    data-testid="loader"
    id="main-loader"
  >
    <span
      class="css-1xdhyk6"
    >
      <span
        class="css-1wl1zro"
      />
      <span
        class="css-nfvbbo"
      />
      <span
        class="css-1xsdpkx"
      />
    </span>
  </div>
</div>
`;

exports[`<Loader> matches snapshot with fullscreen 1`] = `
<div>
  <div
    class="loading-container css-6ju4xn-container-container"
    data-testid="loader"
  >
    <span
      class="css-1xdhyk6"
    >
      <span
        class="css-1wl1zro"
      />
      <span
        class="css-nfvbbo"
      />
      <span
        class="css-1xsdpkx"
      />
    </span>
  </div>
</div>
`;
